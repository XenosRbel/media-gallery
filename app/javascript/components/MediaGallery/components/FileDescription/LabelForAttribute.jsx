import React from 'react';
import PropTypes from 'prop-types';

import { sanitizeAttributeName } from '../../../common/utils';
import { translateFileAttribute } from '../../utils/common';

const LabelForAttribute = ({ name, value }) => (
  <div>
    <span variant="small">{translateFileAttribute(sanitizeAttributeName(name))}</span>
    <span>: </span>
    <span variant="small">{value}</span>
  </div>
);

LabelForAttribute.propTypes = {
  name: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

LabelForAttribute.defaultProps = {
  name: '',
  value: '',
};

export default LabelForAttribute;
