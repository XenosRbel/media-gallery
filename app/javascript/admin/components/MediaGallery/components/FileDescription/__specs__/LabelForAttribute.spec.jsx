import React from 'react';
import { shallow } from 'enzyme';

import LabelForAttribute from '../LabelForAttribute';

describe('Component: LabelForAttribute', () => {
  const wrapper = shallow(<LabelForAttribute />);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
