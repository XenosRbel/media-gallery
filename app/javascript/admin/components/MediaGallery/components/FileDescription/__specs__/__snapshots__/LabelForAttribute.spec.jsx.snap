// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: LabelForAttribute renders correctly 1`] = `
<div>
  <i
    className=""
    code={false}
    copyable={false}
    delete={false}
    disabled={false}
    editable={false}
    ellipsis={false}
    mark={false}
    onChange={[Function]}
    strong={false}
    theme="default"
    type={null}
    underline={false}
    variant="small"
  >
    i18n_mock
  </i>
  <span>
    : 
  </span>
  <i
    className=""
    code={false}
    copyable={false}
    delete={false}
    disabled={false}
    editable={false}
    ellipsis={false}
    mark={false}
    onChange={[Function]}
    strong={false}
    theme="default"
    type={null}
    underline={false}
    variant="small"
  />
</div>
`;
